artiq/firmware/libproto_artiq/session_proto.rs

```diff
@@ -191,9 +192,10 @@ impl<'a> Reply<'a> {
                     let exception = exception.as_ref().unwrap();
                     writer.write_u32(exception.id as u32)?;
                     write_exception_string(writer, &exception.message)?;
-                    writer.write_u64(exception.param[0] as u64)?;
-                    writer.write_u64(exception.param[1] as u64)?;
-                    writer.write_u64(exception.param[2] as u64)?;
+                    writer.write_u64(exception.param.0 as u64)?;
+                    writer.write_u64(exception.param.1 as u64)?;
+                    writer.write_u64(exception.param.2 as u64)?;
+                    writer.write_string(exception.param.3)?;
```

Comment:
All this just so you get a mutable slot in ``params`` ? Or am I missing out something?

Reply:
I don't know where to put the name on when I am not sure where "channel x" is inside the message.